File: Backend/app/services/diagnostico_service.py
from sqlalchemy.orm import Session
from sqlalchemy import or_
from app.models.diagnostico_cie10 import DiagnosticoCIE10

class DiagnosticoService:
    @staticmethod
    def buscar_diagnosticos(db: Session, query: str, limit: int = 20):
        """
        Busca diagnósticos CIE-10 por código o descripción.
        Retorna lista de objetos ORM que serán convertidos por Pydantic.
        """
        if not query or len(query) < 2:
            return []
        
        search_pattern = f"%{query.upper()}%"  # Convertir a mayúsculas para búsqueda
        
        try:
            diagnosticos = db.query(DiagnosticoCIE10).filter(
                or_(
                    DiagnosticoCIE10.codigo.ilike(search_pattern),
                    DiagnosticoCIE10.descripcion.ilike(search_pattern)
                )
            ).limit(limit).all()
            
            # Debug: Verificar qué tipo de objetos estamos devolviendo
            if diagnosticos:
                print(f"✅ Se encontraron {len(diagnosticos)} diagnósticos")
                print(f"   Primer resultado tipo: {type(diagnosticos[0])}")
                print(f"   Primer resultado: id={diagnosticos[0].id}, codigo={diagnosticos[0].codigo}")
            else:
                print(f"⚠️ No se encontraron diagnósticos para: {query}")
            
            return diagnosticos
        except Exception as e:
            print(f"❌ Error al buscar diagnósticos: {e}")
            return []
    
    @staticmethod
    def obtener_por_codigo(db: Session, codigo: str):
        """
        Obtiene un diagnóstico específico por su código CIE-10
        """
        return db.query(DiagnosticoCIE10).filter(
            DiagnosticoCIE10.codigo == codigo
        ).first()

File: Frontend/src/services/diagnosticoService.js
const API_URL = import.meta.env.VITE_API_URL || 'http://localhost:8000';

export const diagnosticoService = {
  /**
   * Busca diagnósticos CIE-10 por código o descripción
   * @param {string} query - Término de búsqueda
   * @param {number} limit - Cantidad máxima de resultados
   * @returns {Promise<Array>}
   */
  async buscar(query, limit = 20) {
    const token = localStorage.getItem('token');
    
    const response = await fetch(
      `${API_URL}/diagnosticos/buscar?query=${encodeURIComponent(query)}&limit=${limit}`,
      {
        headers: {
          'Authorization': `Bearer ${token}`,
          'Content-Type': 'application/json'
        }
      }
    );

    if (!response.ok) {
      throw new Error('Error al buscar diagnósticos CIE-10');
    }

    return await response.json();
  },

  /**
   * Obtiene un diagnóstico específico por su código
   * @param {string} codigo - Código CIE-10
   * @returns {Promise<Object>}
   */
  async obtenerPorCodigo(codigo) {
    const token = localStorage.getItem('token');
    
    const response = await fetch(
      `${API_URL}/diagnosticos/${codigo}`,
      {
        headers: {
          'Authorization': `Bearer ${token}`,
          'Content-Type': 'application/json'
        }
      }
    );

    if (!response.ok) {
      throw new Error('Diagnóstico CIE-10 no encontrado');
    }

    return await response.json();
  }
};
